### ISSUE TITLE
Flexible Attributes created with uppercase characters cannot be searched.

### ISSUE THREAD

**Issue by joshproehl** [beetbox/beets#4565]
It appears that when using flexible attributes the attribute names must be all lower case. (If I had to guess a .lower() is called on the query, but is not called on the string when creating the flexible attribute.)

Here is the debugging steps I went through to discover the issue:

When importing albums I created a tag that had uppercase characters. `beet import . --set Tag_With_Uppercase=true`
I was unable to search for this tag using `beet ls -a Tag_With_Uppercase::.`
After some testing I determined that a lower case version of my tag worked fine, so I went into the SQLite database directly and ran
```sql
UPDATE item_attributes SET key='tag_with_uppercase' WHERE key='Tag_With_Uppercase';
UPDATE album_attributes SET key='tag_with_uppercase' WHERE key='Tag_With_Uppercase';
```
after which finding all my albums using `beet ls -a tag_with_uppercase::.` successfully finds my albums.

I was unable to find where this behavior was documented, so I believe it to be a bug.

**Comment by sampsyo** [comment]
Huh! That's obviously bad, and the root cause is similarly obvious. It's here, when we're parsing queries:
https://github.com/beetbox/beets/blob/d7273bddb34bb247d3ff76f629b8ffa188083cf0/beets/dbcore/queryparse.py#LL147

I'm honestly not sure what the right solution is here, however. We could stop doing this, or we could make the field matching case-insensitive. Probably the latter? I'd be interested in anyone's opinions, if they have any.

**Comment by joshproehl** [comment]
From a user perspective, I'd be fine with either automatically downcasing all tags on creation (and documenting that behavior), or case insensitive search. In either case my preference for tagging systems would be that the tags "joshproehl" and "JoshProehl" should be identical, it's more of a decision about if they're being stored normalized as lowercase or not? 

Arguably since nobody has come across this before means that nobody is trying to use mixed-case tags, so either downcasing tags on creation or case insensitive search should have no impact on existing workflows.

**Comment by sampsyo** [comment]
Yeah, I tend to agree that the solution (regardless of where we implement it) should probably normalize case. It seems real weird for tag names to be case sensitive.

**Comment by gbcox** [comment]
I'm having the exact same issue. I created a flexible tag GENRE_CANON to cleanup my genres without disturbing existing tags, but I consistently can't get it added to the Beets database during import.

I even tried pre-processing my FLAC files with mutagen to force the tag name to genre_canon (lowercase) directly in the file, and then re-importing, but it still didn't help; Beets wouldn't pick it up automatically. beet import --set genre_canon=... works as a workaround, but that's not feasible for a large library.

Also beet info consistently fails to display genre_canon even when beet list genre_canon:value successfully finds the item in the database.

Flexible field functionality is horribly broken.  

beets version 2.3.1
Python version 3.13.5
plugins: chroma, duplicates, info, random, unimported, web
